kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/hs2connection/HiveSiteHS2ConnectionFileParser.java

```diff
@@ -18,42 +18,98 @@
 package org.apache.hive.beeline.hs2connection;
 
 import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
 import java.net.InetAddress;
+import java.net.URI;
 import java.net.URL;
 import java.net.UnknownHostException;
 import java.util.Properties;
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.hive.common.ServerUtils;
-import org.apache.hadoop.hive.conf.HiveConf;
-import org.apache.hadoop.hive.conf.HiveConf.ConfVars;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /*
  * Looks for a hive-site.xml from the classpath. If found this class parses the hive-site.xml
  * to return a set of connection properties which can be used to construct the connection url
  * for Beeline connection
+ *
+ * This class is modified to get rid of dependency on HiveConf
  */
 public class HiveSiteHS2ConnectionFileParser implements HS2ConnectionFileParser {
```

Comment:
it's easy to extend the `HS2ConnectionFileParser` to make the beeline support auto-getting connection URL from `kyuubi-defaults.conf`, we should make such extension in the future.